src/sentry/tasks/commit_context.py

```diff
@@ -0,0 +1,157 @@
+import logging
+from datetime import timedelta
+
+from django.utils import timezone
+
+from sentry.api.serializers.models.release import get_users_for_authors
+from sentry.locks import locks
+from sentry.models import Commit, CommitAuthor, Project, RepositoryProjectPathConfig
+from sentry.models.groupowner import GroupOwner, GroupOwnerType
+from sentry.ownership.grammar import get_source_code_path_from_stacktrace_path
+from sentry.tasks.base import instrumented_task
+from sentry.utils import metrics
+from sentry.utils.event_frames import munged_filename_and_frames
+from sentry.utils.locking import UnableToAcquireLock
+from sentry.utils.sdk import set_current_event_project
+
+PREFERRED_GROUP_OWNERS = 1
+PREFERRED_GROUP_OWNER_AGE = timedelta(days=7)
+
+logger = logging.getLogger("tasks.commit_context")
+
+
+@instrumented_task(
+    name="sentry.tasks.process_commit_context",
+    queue="group_owners.process_commit_context",
+    default_retry_delay=5,
+    max_retries=5,
+)
+def process_commit_context(
+    event_id, event_platform, event_frames, group_id, project_id, sdk_name=None, **kwargs
+):
+    lock = locks.get(
+        f"process-commit-context:{group_id}", duration=10, name="process_commit_context"
+    )
+    try:
+        with lock.acquire():
```

Comment:
The block within this `lock.acquire` seems quite long. I would spend time trying to break it into few functions.